### PR TITLE
Disable checksummed-hex encoding

### LINKED ISSUE #2514: Disable usage of checksummed-hex encoding in the node, but continue to support decoding
Early feedback on the 1.4.3 mainnet release shows that some stakeholders are unable to work with the checksummed-hex encoding as currently implemented.

To resolve this, we should reduce the scope of this encoding to only `AccountHash` and `PublicKey` types.

### PR DESCRIPTION
**Pull request by Fraser999**
This PR disables all instances of checksummed-hex encoding, leaving checksummed-hex decoding unmodified.

This will affect downstream consumers who expect checksummed-hex encoding to be applied to all hash or digest types output by the node or client.

Despite the output types no longer using mixed-case hex, mixed-case hex as input for all such types will continue to have checksum validation performed on them.

Closes #2514.

### CODE REVIEW

**Review by nilz3ro — APPROVED** [review]

File: node/src/components/small_network/message.rs [hunk 1]
```diff
@@ -352,6 +352,24 @@ mod tests {
         55, 55, 53, 48, 66, 67, 67, 69, 65, 69,
     ];
 
+    /// A "conserved" version 1.4.3 handshake.
+    ///
+    /// NEVER CHANGE THIS CONSTANT TO MAKE TESTS PASS, AS IT IS BASED ON MAINNET DATA.
+    const V1_4_3_HANDSHAKE: &[u8] = &[
```
Comment (marc-casperlabs) [inline node/src/components/small_network/message.rs:358]: Is this table-to-farm, i.e. did you acquire this from a live 1.4.3 and dumping it?

**Review by dwerner — APPROVED** [review]
lgtm

**Review by darthsiroftardis — APPROVED** [review]

**Review by marc-casperlabs — COMMENTED** [review]
The networking portions look good at first glance at least. Did not run a full cross version compatibility, I'll leave that to SRE :)

**Comment by Fraser999** [comment]
bors merge

File: node/src/components/small_network/message.rs (hunk 1 above)
Comment (Fraser999) [reply node/src/components/small_network/message.rs:358]: No, I acquired it by running test code from this module, but while checked out to the commit from which 1.4.3 was cut.

**Comment by xcthulhu** [comment]
@Fraser999 I went ahead and created a ticket to reenable this in 2.0

File: node/src/components/small_network/message.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components/small_network/message.rs:358]: That's actually what I meant, oops. Thank you!

**Pull request merged**